### DISCUSSION TITLE
pods_api_pre_save_pod_item_field_event $is_new_item false when adding entry

### DISCUSSION THREAD

**Discussion by HmCody** [pods-framework/pods#6802] [Ideas]
I am using pods_api_pre_save_pod_item_field_event to run some code when a new entry is made for one of my custom pods, field_event, but I am sometimes getting an un-expected value for the $is_new_item parameter.

If I run this code:

```
$newID = $event->duplicate();
$newEvent = pods('field_event', $newID );
$newPostTitle = strval($lastYear + 1) . " Event " .  strval($count + 1);
$data = array(
    'post_title' => $newPostTitle,
    'event_date' => date("Y-m-d H:i", strtotime($event->display('event_date') . " +364 days")),
    'display_date' => date("Y-m-d H:i", strtotime($event->display('display_date') . " +364 days")),
    'registration_opens' => date("Y-m-d H:i", strtotime($event->display('registration_opens') . " +364 days")),
    'registration_closes' => date("Y-m-d H:i", strtotime($event->display('registration_closes') . " +364 days")),
    'linked_post' => '0',
    );
$newEvent->save($data);
```

pods_api_pre_save_pod_item_field_event fires twice, once after the first line, at which point $is_new_item is _true_, and once after the last line, at which point $is_new_item is _false_. This is what I would expect.

If I modify an existing Field Event manually, pods_api_pre_save_pod_item_field_event fires after I hit the "Update" button, and $is_new_item is _false_, as I would expect.

If, however, I add a new Field Event using the "Add New" button on the Field Events page, pods_api_pre_save_pod_item_field_event does not fire immediately, but only after I hit the "Publish" button, at which point  I would expect $is_new_item to be _true_, but it is _false_. 

Is this a bug, or am I missing something about how this should be working?

I am testing this on my local machine using WordPress 5.7.2 and Pods 2.7.27.

ETA: In case it helps, here is the code that uses the filter:

```
add_filter('pods_api_pre_save_pod_item_field_event', 'pre_save_field_event_function', 10, 2);
function pre_save_field_event_function($pieces, $is_new_item) {

    $linked_post_id = $pieces[ 'fields' ][ 'linked_post' ][ 'value' ];

    if ($linked_post_id == 0 or !$is_new_item) {
        //Create or update an associated announcement post

        $eventDate = date('n/j/Y', strtotime($pieces[ 'fields' ][ 'event_date' ][ 'value' ]));
        $postTitle = $eventDate . ' Field Training Day';
        $user_id = get_current_user_id();
        $postDate = date('Y-m-d H:i', strtotime($pieces[ 'fields' ][ 'display_date' ][ 'value' ]));
        $postStatus = 'Publish';
        if ($postDate > date('Y-m-d H:i')) {
            $postStatus = 'Future';
        }

        $postImage = $pieces[ 'fields' ][ 'field_picture' ][ 'value' ];

        $eventID = $pieces[ 'params' ]->id;
        $postContent = '[crfcrc-field-announcement event_id="'  . $eventID .  '"]';

        if ($linked_post_id == 0) {
            //Create a new post
            $field_post = array(
                'post_title'    => $postTitle,
                'post_content'  => $postContent,
                'post_status'   => $postStatus,
                'post_date'     => $postDate,
                'post_author'   => $user_id,
                'post_tag' => array( 'field' ),
                'event_image_gallery' => $postImage,
                'sort_date' => date('Y-m-d H:i', strtotime($pieces[ 'fields' ][ 'event_date' ][ 'value' ])),
            );
            $newID = wp_insert_post($field_post);

            if ($newID > 0 ) {
                $pieces[ 'fields' ][ 'linked_post' ][ 'value' ] = $newID;
            }
        } else {
            //Update the current post
            $field_post = array(
                'ID'            => $linked_post_id,
                'post_title'    => $postTitle,
                'post_content'  => $postContent,
                'post_status'   => $postStatus,
                'post_date'     => $postDate,
                'post_author'   => $user_id,
                'event_image_gallery' => $postImage,
                'sort_date' => date('Y-m-d H:i', strtotime($pieces[ 'fields' ][ 'event_date' ][ 'value' ])),
            );
            wp_insert_post($field_post);
        }

    }

    return $pieces;
}
```

**Comment by pdclark** [comment]
As far as I can tell, [$is_new_item gets defined](https://github.com/pods-framework/pods/blob/main/classes/PodsAPI.php#L3498-L3506) based on whether an ID exists for the post object yet. However, WordPress starts auto-saving the post as soon as the title or any content is entered, which creates an initial database entry.

I'm seeing similar behavior on the `save_post` and `save_post_{$post_type}` hooks, where `$update` is `true` when Publish is initially clicked.

Not sure if it's a solution for you, but I wouldn't rely on `$is_new_item`, and instead would query whether the linked post had already been created or not.

**Comment by HmCody** [comment]
Thanks. That is pretty much the conclusion I had come to and the work around I am using. It just seems...inelegant. And since $is_new_item is used in [Pods documentation](https://docs.pods.io/code-snippets/create-post-title-from-fields-in-the-post-using-pods_api_pre_save/), I don't think this is how the developers are intending for it to work.

**Comment by sc0ttkclark** [comment]
We have some code that tries to help with this. I wish it was more elegant so I think we can try to revisit this again.

**Comment by steven-cheng** [comment]
I still encounter this issue, up to this date. Shouldn't the development team hide this parameter, until it is resolved? I've perplexed by this parameter's abnormal behaviour until I saw this post.
